synapseclient/core/credentials/credential_provider.py

```diff
@@ -97,93 +98,40 @@ def _create_synapse_credential(self, syn, username, password, api_key, auth_toke
 
 class UserArgsCredentialsProvider(SynapseCredentialsProvider):
     """
-    Retrieves auth info from user_login_args
+    Retrieves auth info from user_login_args during a CLI session.
     """
 
-    def _get_auth_info(self, syn, user_login_args):
+    def _get_auth_info(
+        self, syn: "Synapse", user_login_args: Dict[str, str]
+    ) -> Tuple[str, str]:
         return (
             user_login_args.username,
-            user_login_args.password,
-            user_login_args.api_key,
             user_login_args.auth_token,
         )
 
 
-@deprecated.sphinx.deprecated(
-    version="1.9.0",
-    action="ignore",
-    reason="This will be removed in 2.0. Please use username and password or apiKey instead.",
-)
-class UserArgsSessionTokenCredentialsProvider(SynapseCredentialsProvider):
-    """
-    This is a special case where we are not given context as to what the username is. We are only given a session token
-    and must retrieve the username and api key from Synapse
-    """
-
-    def _get_auth_info(self, syn, user_login_args):
-        username = None
-        password = None
-        api_key = None
-        auth_token = None
-
-        if user_login_args.session_token:
-            username = syn.getUserProfile(sessionToken=user_login_args.session_token)[
-                "userName"
-            ]
-            api_key = syn._getAPIKey(user_login_args.session_token)
-
-        return username, password, api_key, auth_token
-
-
 class ConfigFileCredentialsProvider(SynapseCredentialsProvider):
     """
-    Retrieves auth info from .synapseConfig file
+    Retrieves auth info from the `~/.synapseConfig` file
     """
 
-    def _get_auth_info(self, syn, user_login_args):
+    def _get_auth_info(
+        self, syn: "Synapse", user_login_args: Dict[str, str]
+    ) -> Tuple[Union[str, None], Union[str, None]]:
         config_dict = syn._get_config_authentication()
         # check to make sure we didn't accidentally provide the wrong user
 
         username = config_dict.get("username")
-        password = config_dict.get("password")
-        api_key = config_dict.get("apikey")
         token = config_dict.get("authtoken")
 
         if user_login_args.username and username != user_login_args.username:
-            # if the username is provided and there is a config file username but they don't match
-            # then we don't use any of the values from the config to prevent ambiguity
+            # if the username is provided and there is a config file username but they
+            # don't match then we don't use any of the values from the config
+            # to prevent ambiguity
```

Comment:
Should this log a warning?

Reply:
We might remove username totally. See: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1041#discussion_r1444892183

If we don't I'll add in a log here.

Reply:
I added a warning in latest commit